contracts/HAVToken.sol

```diff
@@ -5,127 +5,106 @@
     return withinCap;
   }
 
-  // fallback function to buy tokens
   function() payable {
     buyHAVTokens(msg.sender);
   }
 
-  function buyHAVTokens(address beneficiary) payable {
-    require(beneficiary != 0x0);
+  function buyHAVTokens(address recip) payable {
+    require(recip != 0x0);
     require(approvePurchase());
 
-    uint256 weiAmount = msg.value;
-    uint256 numHAVTokens = weiAmount.mul(HAVPerEther);
-    cumEtherRaised = cumEtherRaised.add(weiAmount);
+    uint256 numWei = msg.value;
+    uint256 numHav = numWei.mul(rate);
+    etherRaised = etherRaised.add(numWei);
 
-    issueHAVForDID(beneficiary, numHAVTokens);
-    HAVPurchase(msg.sender, weiAmount, numHAVTokens);
+    issueHAVForDID(recip, numHav);
+    LogHAVSale(msg.sender, numWei, numHav);
   }
 
-  function issueHAVForDID(address _to, uint256 _amount) internal returns (bool) {
-
-    // Reduce DID held prior to issuing HAV to prevent reentrancy (CRUCIAL/SECURITY)
-    DIDToken didToken = DIDToken(DIDTokenAddress);
-    bool burnDIDSuccess = didToken.exchangeDIDForHAV(_to, _amount);
-    if (burnDIDSuccess) {
-      numHAVOutstanding = numHAVOutstanding.add(_amount);
-      balancesHAV[_to] = balancesHAV[_to].add(_amount);
-      DIDHAVExchange(_to, _amount);
+  function issueHAVForDID(address _to, uint256 _num) internal returns (bool) {
+    // Prevent reentrancy -- SECURITY
+    DIDToken didToken = DIDToken(DIDAddress);
+    bool burntDID = didToken.burnDIDHAV(_to, _num);
+    if (burntDID) {
+      numHAV = numHAV.add(_num);
+      bals[_to] = bals[_to].add(_num);
+      DIDHAV(_to, _num);
       return true;
     }
     else {
       return false;
     }
   }
 
-  //  send ether to the fund collection wallet
-  //  override to create custom fund forwarding mechanisms
   function forwardFunds() internal {
     //    TODO wallet.transfer(msg.value);
   }
 
-  // @return true if the transaction can buy tokens
   function approvePurchase() internal constant returns (bool) {
-    bool nonZeroPurchase = msg.value != 0;
-    return nonZeroPurchase && saleActive;
+    return msg.value != 0 && saleOn;
   }
 
-  //  If a trading market exists we need to update a sale price
-  function tradingMarketExists(bool _tradingMarketExists) public returns (bool) {
+  function tradMkt(bool _tradMkt) public returns (bool) {
     require(msg.sender == owner);
-    tradingMarketExists = _tradingMarketExists;
+    tradMkt = _tradMkt;
   }
 
   function numberAvailableForSale() constant returns (uint256) {
-    return maxBalanceEther - currentBalanceEther;
+    return maxEther - etherBal;
   }
 
   function pauseSale() public returns (bool) {
     require(msg.sender == owner);
-    saleActive = !saleActive;
-    LogHAVSaleStatusChange(saleActive);
+    saleOn = !saleOn;
+    LogStatusChange(saleOn);
     return true;
   }
 
-  modifier onlyDIDTokenAddress() {
-    require(msg.sender == DIDTokenAddress);
+  modifier onlyDIDAddress() {
+    require(msg.sender == DIDAddress);
```

Comment:
I didn't event know about `approveAddress(DIDAddress)` when I wrote this so will change.